devices/MAX1704x/Max1704X.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.MAX1704x
+{
+    /// <summary>
+    /// Base class for controlling MAX1704x via I2C interface.
+    /// </summary>
+    public abstract class Max1704X
+    {
+        /// <summary>
+        /// The default I2C address.
+        /// </summary>
+        public const byte DefaultAddress = 0x36;
+        
+        private const byte ThresholdMinValue = 32;
+
+        /// <summary>
+        /// Gets or sets the compensation value for the model gauge algorithm.
+        /// The upper 8 bits of the CONFIG register control the compensation.
+        /// </summary>
+        /// <value>
+        /// The compensation value as a byte.
+        /// </value>
+        public byte Compensation
+        {
+            get
+            {
+                var configReg = GetConfigRegister();
+                var compensation = (configReg & 0xFF00) >> 8;
+                return (byte)compensation;
+            }
+
+            set
+            {
+                var configReg = GetConfigRegister();
+                configReg &= 0x00FF;
+                configReg |= value;
+                Write16(configReg, Registers.Max17043Config);
+            }
+        }
+
+        /// <summary>
+        /// Gets the voltage of the battery.
+        /// </summary>
+        /// <returns>The voltage of the battery in volts.</returns>
+        public float BatteryVoltage
+        {
+            get
+            {
+                var vCell = Read16(Registers.Max17043Vcell);
+                return vCell / Divider * FullScale;    
+            }
+        }
+
+        /// <summary>
+        /// Gets the battery percentage.
+        /// </summary>
+        /// <returns>The battery percentage as a float value.</returns>
+        public float BatteryPercent
+        {
+            get
+            {
+                var soc = Read16(Registers.Max17043Soc);
+                var percent = (float)((soc & 0xFF00) >> 8);
+                percent += (soc & 0x00FF) / 256.0f;
+                return percent;    
+            }
+        }
+
+        /// <summary>
+        /// Gets the version of the device.
+        /// </summary>
+        /// <value>The version of the device.</value>
+        public ushort Version => Read16(Registers.Max17043Version);
+
+        /// <summary>
+        /// Gets or sets the threshold of the property.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the value is out of range.</exception>
+        public byte Threshold
```

Comment:
what is the threshold represents? Also please mention the range in the intellisense comments.

Reply:
done